docs/assignment3.md

```diff
@@ -1 +1,306 @@
+
+The team editor checks if the player selection is correct, as shown in the Alt fragment. If there are not enough players to create a team, showError() is called and displays an error message showing which fields are empty.
+If the same player appears multiple times in the selection, showError() is called and an error message is displayed. If an error occurs, createTeam() exits and the player must press "Save Team" to start again.
+
+If there are no errors, addTeam() is called, and :Game creates a new team object and adds it to the team pool.
+
+`Sequence diagram showing League creation and execution`
+![](imgs/Sequence2League.png)
+
+This sequence diagrams shows the interactions between the player and the game to execute the league from a player provided schedule. This sequence happens when all teams have been made and the player starts the league simulation. 
+The player represented by the actor element provides a schedule with includes the number of matches and which teams will play in the first round.
+
+Once the Start button is pressed, the league simulator checks for errors as shown in the alt fragment. If there are matches missing teams, showError is called and an error message is displayed. This causes the league execution to stop and the player has to hit the start button again.
+ 
+If there are no errors, the League object is created and the execution of the league begins.
+The league starts with the playRound function. Here all the scheduled matches are played one by one in the Loop fragment. A match object is created and returns the winner of the match before being destroyed. The winning teams are added to the pool of teams playing next round. 
+
+Once a round is over, the Alt fragment checks if there are more than one remaining team. If yes playRound is called recursively, as shown by the self-call element. The exit condition is one team remaining, and when it is met the results are sent to the game and the league object is destroyed.
+The full results are then displayed by LeagueSimulatorPane.
+
+
+## Implementation									
+Author(s): `Teona Moga`
+
+
+`Video executing the program:`
+
+![](imgs/gif.gif)
+
+The first step in our implementation process was coming up with very basic UML diagrams that outlined the basic classes and functionality required by the system. We followed an Agile approach, where we would go back and forth between our code and the diagrams, until we achieved the desired functionality.  
+
+When starting Assignment 3, we focused on implementing the remaing requirements that would result in a complete game experience for the user, such that now the system supports team creation and displays the leaderboard. The only major change is  how player data is generated - we first intended for the user to upload player data, but later decided to let the system do it, in order to make it more similar to the real experience. 
```

Comment:
**[SEVERE]** A key aspect of this last assignment is to explain the rationale behind your addition or removal of features. Not only do I not see an explanation for removing F4 : Multiple teams, but you decided to change an important aspect of the game engine and justify it with a short and brief sentence. in what ways is it more similar to the real experience. Lacking and unconvincing explanation!